sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/batch/TransactionalBatchResponse.java

```diff
@@ -0,0 +1,514 @@
+    public int indexOf(Object o) {
+        return this.results.indexOf(o);
+    }
+
+    @Override
+    public Iterator<TransactionalBatchOperationResult<?>> iterator() {
+        return this.results.iterator();
+    }
+
+    @Override
+    public int lastIndexOf(Object o) {
+        return 0;
+    }
+
+    @Override
+    public ListIterator<TransactionalBatchOperationResult<?>> listIterator() {
+        return null;
+    }
+
+    @Override
+    public ListIterator<TransactionalBatchOperationResult<?>> listIterator(int index) {
+        return null;
+    }
+
+    @Override
+    public boolean remove(Object result) {
+        return this.results.remove(result);
+    }
+
+    @Override
+    public TransactionalBatchOperationResult<?> remove(int index) {
+        return null;
+    }
+
+    @Override
+    public boolean removeAll(Collection<?> collection) {
+        return this.results.removeAll(collection);
+    }
+
+    @Override
+    public boolean retainAll(Collection<?> collection) {
+        return this.results.retainAll(collection);
+    }
+
+    @Override
+    public TransactionalBatchOperationResult<?> set(int index, TransactionalBatchOperationResult<?> result) {
+        return this.results.set(index, result);
+    }
+
+    @Override
+    public List<TransactionalBatchOperationResult<?>> subList(int fromIndex, int toIndex) {
+        return this.results.subList(fromIndex, toIndex);
+    }
+
+    @Override
+    public Object[] toArray() {
+        return this.results.toArray();
+    }
+
+    @Override
+    public <T> T[] toArray(T[] a) {
+        return this.results.toArray(a);
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.results.isEmpty();
+    }
+
+    @Override
+    public boolean add(TransactionalBatchOperationResult<?> result) {
+        return this.results.add(result);
+    }
+
+    @Override
+    public void add(int index, TransactionalBatchOperationResult<?> element) {
+        this.results.add(index, element);
+    }
+
+    @Override
+    public boolean addAll(Collection<? extends TransactionalBatchOperationResult<?>> collection) {
+        return this.results.addAll(collection);
+    }
+
+    @Override
+    public boolean addAll(int index, Collection<? extends TransactionalBatchOperationResult<?>> collection) {
+        return this.results.addAll(index, collection);
+    }
+
+    @Override
+    public void clear() {
+        this.results.clear();
+    }
+
+    @Override
+    public boolean contains(Object result) {
+        return this.results.contains(result);
+    }
+
+    @Override
+    public boolean containsAll(Collection<?> c) {
+        return false;
+    }
+
+    /**
+     * Closes the current {@link TransactionalBatchResponse}.
+     */
+    public void close() {
+        if (!this.isDisposed) {
+            this.isDisposed = true;
+            if (operations != null) {
+                for (ItemBatchOperation<?> operation : operations) {
+                    operation.close();
```

Comment:
try catch when closing each item?

if one close throws, the loop will terminate.

Reply:
Operation.close() function catches the exception if any. So this should be fine.